actions/cleanup-old-branches/scripts/cleanup-old-branches.sh

```diff
@@ -156,66 +148,25 @@ function should_keep_branch() {
 }
 
 # Get list of head/feature branches that are associated with open pull requests.
-function fetch_open_pr_branches() {
+function get_open_pr_branches() {
   # Clear global array before filling it.
   OPEN_PR_BRANCHES=()
-  # Initialize cursor for pagination.
-  local end_cursor=""
-
-  # Loop to fetch all pages of pull requests.
-  while :; do
-    # Fetch current batch of PRs and update cursor.
-    response=$(gh api graphql -f query="
-      query(\$endCursor: String) {
-          repository(owner: \"$GITHUB_REPOSITORY_OWNER\", name:\"$GITHUB_REPOSITORY_NAME\") {
-              pullRequests(states: OPEN, first: 100, after: \$endCursor) {
-                  nodes {
-                      headRefName
-                  }
-                  pageInfo {
-                      hasNextPage
-                      endCursor
-                  }
-              }
-          }
-      }" -f endCursor="$end_cursor" --paginate)
-    
-    # Check if there are PR branches to process.
-    if ! echo "${response}" | jq -e '.data.repository.pullRequests.nodes | length > 0' >/dev/null; then
-        echo "::debug::No PR branches found in the current fetch."
-    else
-        # Append PR head branch names to the OPEN_PR_BRANCHES array.
-        while IFS= read -r line; do
-            OPEN_PR_BRANCHES+=("$line")
-        done < <(echo "$response" | jq -r '.data.repository.pullRequests.nodes[].headRefName')
-    fi
-
-    head_branch_names=$(echo "$response" | jq -r '.data.repository.pullRequests.nodes[].headRefName')
-    readarray -t OPEN_PR_BRANCHES <<< "$head_branch_names"
-
-    # Print the current batch of PR branches.
-    if [ "${#OPEN_PR_BRANCHES[@]}" -gt 0 ]; then
-      echo "::debug::Current batch of PR Branches: ${OPEN_PR_BRANCHES[*]}"
-    else
-      echo "::debug::No PR branches to display."
-    fi
-
-    # Check if there are more pages.
-    has_next_page=$(echo "$response" | jq -r '.data.repository.pullRequests.pageInfo.hasNextPage')
-    if [[ "${has_next_page}" == "false" ]]; then
-      echo "::debug::No more pages to fetch on fetch_open_pr_branches()."
-      break
-    fi
-
-    # Update cursor for next page
-    end_cursor=$(echo "$response" | jq -r '.data.repository.pullRequests.pageInfo.endCursor')
-  done
+  local pr_branch_names
+  # 1,000 PRs is the hard limit.
+  pr_branch_names=$(gh pr list --state open \
+    --json headRefName --jq '.[].headRefName' --limit 1000)
+  readarray -t OPEN_PR_BRANCHES <<< "${pr_branch_names}"
+
+  if [[ "${#OPEN_PR_BRANCHES[@]}" -eq 1000 ]]; then
+    echo "::error::Too many open PR Branches. Limit of 1,000 open PRs reached: ${OPEN_PR_BRANCHES[*]}"
```

Comment:
Should we really exit here? I feel like it should be fine to just batch process and warn that not all branches were processed.

Reply:
Discussing in a side channel.